packages/compass-query-bar/src/components/query-history-button-popover.tsx

```diff
@@ -0,0 +1,136 @@
+import type AppRegistry from 'hadron-app-registry';
+
+const { track } = createLoggerAndTelemetry('COMPASS-QUERY-BAR-UI');
+
+const openQueryHistoryStyles = cx(
+  css({
+    border: 'none',
+    backgroundColor: 'transparent',
+    display: 'inline-flex',
+    alignItems: 'center',
+    padding: spacing[2] - 2, // -2px for border.
+    '&:hover': {
+      cursor: 'pointer',
+    },
+    '&:focus': focusRingVisibleStyles,
+  }),
+  focusRingStyles
+);
+
+const queryHistoryContainerStyles = css({
+  display: 'flex',
+  height: '100%',
+});
+
+const queryHistoryPopoverStyles = css({
+  // We want the popover to open almost to the shell at the bottom of Compass.
+  maxHeight: 'calc(100vh - 270px)',
+  display: 'flex',
+  marginLeft: -spacing[4], // Align to the left of the query bar.
+});
+
+type QueryHistoryProps = {
+  globalAppRegistry: AppRegistry;
+  localAppRegistry: AppRegistry;
+};
+
+export const QueryHistoryButtonPopover: React.FunctionComponent<
+  QueryHistoryProps
+> = ({ globalAppRegistry, localAppRegistry }) => {
+  const queryHistoryRef = useRef<{
+    component: React.ComponentType<any>;
+    store: any; // Query history store is not currently typed.
+    actions: any; // Query history actions are not typed.
+  }>({
+    component: globalAppRegistry.getRole('Query.QueryHistory')![0].component,
+    store: localAppRegistry.getStore('Query.History'),
+    actions: localAppRegistry.getAction('Query.History.Actions'),
+  });
+  const queryHistoryButtonRef = useRef<HTMLButtonElement>(null);
+  const queryHistoryContainerRef = useRef<HTMLDivElement>(null);
+
+  const [showQueryHistory, setShowQueryHistory] = useState(false);
+
+  const onClickQueryHistory = useCallback(() => {
+    if (!showQueryHistory) {
+      track('Query History Opened');
+    }
+
+    setShowQueryHistory(!showQueryHistory);
+  }, [showQueryHistory, setShowQueryHistory]);
+
+  const onClickOutsideQueryHistory = useCallback(
+    (event) => {
+      // Ignore clicks on the query history button as it has its own handler.
+      if (
+        !queryHistoryButtonRef.current ||
+        queryHistoryButtonRef.current.contains(event.target as Node)
+      ) {
+        return;
+      }
+      setShowQueryHistory(false);
+    },
+    [queryHistoryButtonRef, setShowQueryHistory]
+  );
+
+  useOnClickOutside(queryHistoryContainerRef, onClickOutsideQueryHistory);
+
+  const onHideQueryHistory = useCallback(() => {
+    setShowQueryHistory(false);
+  }, [setShowQueryHistory]);
+
+  const QueryHistoryComponent = queryHistoryRef.current?.component;
+
+  return (
+    <>
+      <button
+        data-testid="query-history-button"
+        onClick={onClickQueryHistory}
+        className={openQueryHistoryStyles}
+        id="open-query-history"
+        aria-label="Open query history"
+        type="button"
+        ref={queryHistoryButtonRef}
+      >
+        <Icon glyph="Clock" />
+        <Icon glyph="CaretDown" />
+      </button>
+      <Popover
```

Comment:
This popover currently is not accessible. There is no exact aria example that would match this that I can link you here to, but basically what you have here is a composite element with an active descendant that is a popup that should manage its own focus, similar to an [action menu](https://www.w3.org/WAI/ARIA/apg/example-index/menu-button/menu-button-actions-active-descendant) or a [toolbar](https://www.w3.org/WAI/ARIA/apg/example-index/toolbar/toolbar.html) (actually somewhere in between the two). At the very least it should assume focus on open and return it back to the trigger on close

Reply:
Created a new component in `compass-components` called `InteractivePopover`. It abstracts the keyboard actions, rendering, and focus trap. If it looks good to you we'll also use this for rendering the saved aggregations list in COMPASS-5852
Let me know how that implementation looks, I'm adding tests now.
It uses `react-focus-trap` to take focus, and now we return focus to the trigger button when the popover is hidden.